### PR TITLE
Use Docsy friendly footer

### PR DESCRIPTION
**Pull request by sftim**
This helps with https://github.com/kubernetes/website/issues/41171

Make the site footer look more like the footer from Docsy.

/area web-development

[Before](https://k8s.io/docs/) | [After](https://deploy-preview-47789--kubernetes-io-main-staging.netlify.app/docs/)

### CODE REVIEW

**Comment by salaxander** [comment]
Deploy preview is looking good so far!

**Comment by reylejano** [comment]
The changes look good
/approve

**Comment by Shubham82** [comment]
Hi @sftim 
Small nit: The footer now shows the `StackOverflow` icon just below the Contribute icon. Can we mention this after the Contribute icon(like others mention in one row)?
Other than that it looks good to me.
WDYT?
Screenshot

**Comment by sftim** [comment]
>  Small nit: The footer now shows the StackOverflow icon just below the Contribute icon. Can we mention this after the Contribute icon(like others mention in one row)?

The old and new footers do look different.

The icons showing across two rows is due to wrapping. I think the wrapping depends on page width. I'm not sure if you're asking to optimize for a particular viewport width (if so, please say what number of pixels and maybe also what DPI), or if it's actually OK now that you know wrapping is the cause.

**Comment by sftim** [comment]
> I can see in the after change URL https://deploy-preview-47789--kubernetes-io-main-staging.netlify.app/docs/home/ now there is slight right to left scrolling option maybe some footer element is extending width. Current URL doesnt show this behavir. Below added screenshot (look in the right part there is white space after the black background navbar.)

Any ideas about how to fix that? I'd love to see progress on upgrading Docsy; the version we use isn't even compatible with current Hugo.

Help is welcome; I definitely don't intend to do the entire Docsy upgrade piece on my own.

File: layouts/partials/footer.html [hunk 1]
```diff
@@ -1,52 +1,44 @@
...
-      {{ with site.GetPage "page" "docs/tutorials/stateless-application/hello-minikube" }}<a href="{{ .RelPermalink }}">{{ .LinkTitle }}</a>{{ end }}
-      {{ $sections := slice "docs/home" "blog" "training" "partners" "community" "case-studies" }}
-      {{ range $sections }}
-      {{ with site.GetPage "section" . }}<a class="text-white" href="{{ .RelPermalink }}">{{ .LinkTitle }}</a>{{ end }}
-      {{ end }}
-    </nav>
-  </div>
-  <div class="container-fluid">
+<footer class="bg-dark py-5 row d-print-none">
+  <div class="container-fluid mx-sm-5">
```
Comment (dipesh-rawat) [inline layouts/partials/footer.html:3]: It looks like the `row` class is adding extra margins to the footer, causing the overflow and scroll issue mentioned in review comment https://github.com/kubernetes/website/pull/47789#pullrequestreview-2284394122.
Below is a suggestion that worked based on my testing using the browser's developer tools.

```suggestion
<footer class="bg-dark py-5 d-print-none">
  <div class="container-fluid">
```

Comment (sftim) [reply layouts/partials/footer.html:3]: Might be the wrong fix, though. `col` class should only be used as a child of something with `row`, I think.

Comment (dipesh-rawat) [reply layouts/partials/footer.html:3]: I thought that the `col` class elements were already nested inside a `row` class div. I haven't had a chance to extensively test this layout.

https://github.com/kubernetes/website/blob/ff933be420f1e86f006a1c6a3599dd7ea392c4e0/layouts/partials/footer.html#L2-L4

**Comment by Shubham82** [comment]
> > Small nit: The footer now shows the StackOverflow icon just below the Contribute icon. Can we mention this after the Contribute icon(like others mention in one row)?
> 
> The old and new footers do look different.
> 
> The icons showing across two rows is due to wrapping. I think the wrapping depends on page width. I'm not sure if you're asking to optimize for a particular viewport width (if so, please say what number of pixels and maybe also what DPI), or if it's actually OK now that you know wrapping is the cause.

@sftim, I checked by changing the width of the browser page from 100% to 90% and it shows the icons in one row.
Changing_width_Screenshot

Other than that I also checked by changing the heading in the **footer-links-block** under the footer.html code, which also shows the icons in one row for me (page width 100%).
https://github.com/kubernetes/website/blob/56777ed2c46daa8f1fd8d6f2728559d89df104a2/layouts/partials/footer.html#L45

I changed the above to the below one (h3->h4): 

`<li class="list-inline-item mx-2 h4" data-toggle="tooltip" data-placement="top" title="{{ .name }}" aria-label="{{ .name }}">`

Code_change_Screenshot

File: assets/scss/_custom.scss [hunk 2]
```diff
@@ -266,16 +266,26 @@ body.td-404 main .error-details {
...
     @media only screen and (min-width: 768px) {
       max-width: calc(min(80rem,90vw)); // avoid spreading too wide
     }
+    color: inherit;
+    background: transparent;
+
+    a:hover {
+      color: inherit;
+      background: transparent;
+      text-decoration: underline;
```
Comment (tengqm) [inline assets/scss/_custom.scss:287]: where does this come from?

File: assets/scss/_custom.scss [hunk 3]
```diff
@@ -313,6 +323,11 @@ footer {
   }
 }
 
+footer {
+  background-image: linear-gradient(rgba(0, 0, 0, 0.527),rgba(0, 0, 0, 0.5)) , url("/images/texture.png");
+  color: #e9e9e9;
+}
```
Comment (tengqm) [inline assets/scss/_custom.scss:329]: where does this come from?

**Comment by sftim** [comment]
I'll try the fix from https://github.com/kubernetes/website/pull/47789#issuecomment-2337855132

**Comment by sftim** [comment]
Here's a different fix.

**Comment by sftim** [comment]
@steve-hardman & @dipesh-rawat your feedback on this was helpful in spotting a snag. Would you be willing to confirm it now looks OK / good enough to merge?

File: assets/scss/_custom.scss (hunk 3 above)
Comment (sftim) [reply assets/scss/_custom.scss:329]: This is my stylistic choice. It styles the site footer without any layout-level override; it's a blend of on our look before we used Docsy, and Docsy's own look (solid dark grey background to the footer).

If we drop using `texture.png`, we can change the page header and footer at the same time. That'd be a future PR.

**Review by dipesh-rawat — COMMENTED** [review]
The footer in deploy [preview](https://deploy-preview-47789--kubernetes-io-main-staging.netlify.app/docs/home/) now looks good. IMO this is okay to merge.

/lgtm

**Pull request merged**